### PR TITLE
Check newer docker-java

### PR DESCRIPTION
**Pull request by KostyaSha**

### CODE REVIEW

**Comment by rnorth** [comment]
Nice, thanks @KostyaSha! It looks like we have some compilation failures (modifies API?) but we can look into these. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by rnorth** [comment]
@KostyaSha I think this makes sense, and I've been able to refactor our code so that it compiles and works again 😄.

Basically, anywhere where we currently just modify a `CreateContainerCmd` we just need to modify a `HostConfig` object as well.

I'd just like to have a bit more of a play with this over the weekend, because this might overlap a little bit with some separate changes we were thinking of making, but it'll probably be fine.

Thanks, BTW, for letting us know proactively about this change! 🙇

**Comment by rnorth** [comment]
I've pushed some changes to this PR branch - I hope that's OK with you. Quite a few changes, but no big deal really!

**Comment by KostyaSha** [comment]
cool, thanks. People started sending PRs for new bugfixes so i will do new releases soon

File: docs/examples/src/test/java/generic/CmdModifierTest.java [hunk 1]
```diff
@@ -20,8 +21,8 @@
     // memory {
     @Rule
     public GenericContainer memoryLimitedRedis = new GenericContainer<>("redis:3.0.2")
-            .withCreateContainerCmdModifier(cmd -> cmd.withMemory((long) 4 * 1024 * 1024))
-            .withCreateContainerCmdModifier(cmd -> cmd.withMemorySwap((long) 4 * 1024 * 1024));
+            .withCreateContainerCmdModifier(cmd -> cmd.withHostConfig(new HostConfig().withMemory((long) 4 * 1024 * 1024)))
```
Comment (KostyaSha) [inline docs/examples/src/test/java/generic/CmdModifierTest.java]: `for new you can use com.github.dockerjava.api.model.HostConfig#newHostConfig`

**Pull request closed without merging**